scipy/sparse/csgraph/_euclidean_min_spanning_tree.py

```diff
@@ -0,0 +1,62 @@
+import numpy as np
+
+from scipy.linalg import norm
+from scipy.sparse import csr_matrix
+from scipy.sparse.csgraph import minimum_spanning_tree
+from scipy.spatial import Delaunay
+from scipy.spatial.distance import squareform, pdist
+
+
+def _delaunay_to_euclidean_graph(tri):
+    """TODO"""
+
+    npoints, ndim = tri.points.shape
+
+    # edge to weight dict
+    edge_to_weight = {}
+    for simplex in tri.simplices:
+        for i in range(-1, ndim):
+            node_from, node_to = sorted(
+                [simplex[i], simplex[i+1]]
+            )  # for undirected graph
+            if (node_from, node_to) in edge_to_weight:
+                continue
```

Comment:
This implementation is not quite right--you are missing quite a few edges in this search. I can make the 4-D example that is failing for you work just fine with a more exhaustive edge search (though you should try to find a more efficient approach--I'm just trying to help you get unstuck):

```diff
--- a/scipy/sparse/csgraph/_euclidean_min_spanning_tree.py
+++ b/scipy/sparse/csgraph/_euclidean_min_spanning_tree.py
@@ -1,4 +1,5 @@
 import numpy as np
+import itertools
 
 from scipy.linalg import norm
 from scipy.sparse import csr_matrix
@@ -15,9 +16,9 @@ def _delaunay_to_euclidean_graph(tri):
     # edge to weight dict
     edge_to_weight = {}
     for simplex in tri.simplices:
-        for i in range(-1, ndim):
+        for tup in itertools.permutations(simplex, 2):
             node_from, node_to = sorted(
-                [simplex[i], simplex[i+1]]
+                list(tup)
             )  # for undirected graph
             if (node_from, node_to) in edge_to_weight:
                 continue
```